### PR TITLE
JSSDK-18- Updating Messages Service

### LINKED ISSUE #18: Update JS SDK for Messages

### PR DESCRIPTION
**Pull request by wambui-pixel**
Pull request title should be `JSSDK-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](./CONTRIBUTING.md) file for detailed contributing guidelines.

### What does this do?
Update messages services to use typescript
### Which issue(s) does this PR fix/relate to?
Resolves #18

Put here `Resolves #XXX` to auto-close the issue that your PR fixes (if such)

### List any changes that modify/break current functionality
N/A
### Have you included tests for your changes?
Yes
### Did you document any new/modified functionality?
N/A

### CODE REVIEW

File: src/messages.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      throw error
+    }
+  }
+
+  public async Read (queryParams: QueryParams, channelId: string, token: string): Promise<MessagesPage> {
+    // Read messages
+    /**
+     *
+     * @method Read - Read messages from a given channel. Messages are read from a reader
+     * add-on such as timescale. Messages are read from a http port specific to the reader
```
Comment (dborovcanin) [inline src/messages.ts:90]: ...such as Timescale.

File: src/messages.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+     * @param {string} channel_id - The channel_id of the channel to read the message from.
```
Comment (dborovcanin) [inline src/messages.ts:91]: ...The ID of the channel to read messages from

File: src/messages.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+     * @param {string} token - The token to be used for authentication.
```
Comment (dborovcanin) [inline src/messages.ts:92]: Authentication token.

File: src/messages.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    channelId: string,
+    msg: string,
+    thingKey: string
+  ): Promise<Response> {
+    // Send a message
+    /**
+     * @method Send- Sends message to a given channel via HTTP protocol. Message is sent
+     * through a writer add-on such as timescale. Message is sent to a
+     * http port specific to the writer add-on. The thing and channel must be
+     * created before sending the message and connected.
```
Comment (dborovcanin) [inline src/messages.ts:42]: The Thing and Channel must exist and be Thing must be connected to the Cahnnel.

File: src/messages.ts [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+  public async Send (
+    channelId: string,
+    msg: string,
+    thingKey: string
+  ): Promise<Response> {
+    // Send a message
+    /**
+     * @method Send- Sends message to a given channel via HTTP protocol. Message is sent
+     * through a writer add-on such as timescale. Message is sent to a
```
Comment (dborovcanin) [inline src/messages.ts:40]: Message is sent to the HTTP adapter.

File: src/messages.ts [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  type Response,
+  type MessagesPage,
+  type QueryParams
+} from './defs'
+
+export default class Messages {
+  // Messages API Client
+  /**
+   * @method Messages - Messages is used to manage messages.
+   * It provides methods for sending and reading messages.
```
Comment (dborovcanin) [inline src/messages.ts:13]: Messages class is used for sending and reading messages.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: src/messages.ts [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    const options: RequestInit = {
+      method: 'GET',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Bearer ${token}`
+      }
+    }
+    try {
+      const response = await fetch(
+
```
Comment (dborovcanin) [inline src/messages.ts:114]: Remove the empty line and align the style with the rest of the codebase.

File: examples/messages.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+const mySdk = new SDK({
+  httpadapterUrl: defaultUrl + ':8008',
+  readersUrl: defaultUrl + ':9011'
+})
+
+const token = '<token>'
+
+mySdk.messages
+  .Send(
+    'channelId',
```
Comment (ianmuchyri) [inline examples/messages.ts:15]:
```suggestion
    '<channelId>',
```

File: examples/messages.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    'message',
```
Comment (ianmuchyri) [inline examples/messages.ts:16]:
```suggestion
    '<message>',
```

File: examples/messages.ts [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    'thingKey'
```
Comment (ianmuchyri) [inline examples/messages.ts:17]:
```suggestion
    '<thingKey>'
```

**Review by dborovcanin — APPROVED** [review]

File: src/messages.ts [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+      subtopic = chanNameParts[1].replace('.', '/')
+    }
+
+    const options: RequestInit = {
+      method: 'POST',
+      headers: {
+        'Content-Type': this.contentType,
+        Authorization: `Thing ${thingKey}`
+      },
+      body: JSON.stringify(new TextEncoder().encode(msg))
```
Comment (ianmuchyri) [inline src/messages.ts:63]:
```suggestion
      body: msg
```

File: src/messages.ts [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+    }
+    try {
+      const response = await fetch(
+        new URL(
+          `channels/${chanId}/messages/${subtopic}`,
```
Comment (ianmuchyri) [inline src/messages.ts:68]:
```suggestion
          `channels/${chanId}/messages${subtopic}`,
```

**Review by ianmuchyri — CHANGES_REQUESTED** [review]

**Review by ianmuchyri — APPROVED** [review]
LGTM

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**